feat: MultichainAccountConnectMultiselector

montelaidev opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR adds the BIP44 `MultichainAccountConnectMultiSelector` component. It is currently not wired up to be used. It can only be tested with storybook. 

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: null

## **Related issues**

Related to: https://consensyssoftware.atlassian.net/browse/MUL-669?atlOrigin=eyJpIjoiODdiMGYzODIxMjJmNDY3M2JlMWVhYjE0MTEwMTcwNmYiLCJwIjoiaiJ9

## **Manual testing steps**

To test use storybook. 

## **Screenshots/Recordings**

### **After**

<!-- [screenshots/recordings] -->
<img width="1179" height="2556" alt="Simulator Screenshot - iPhone 15 Pro - 2025-08-27 at 17 02 14" src="https://github.com/user-attachments/assets/86d94750-3236-4597-a466-e21564d90004" />


## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

montelaidev committed:
feat: add MultichainAccountConnectMultiSelector component and related stories/tests

- Introduced `MultichainAccountConnectMultiSelector` for connecting multiple multichain accounts.
- Added mock store setup for testing and integrated Redux for state management.
- Created stories for the new component in Storybook, showcasing various states and interactions.
- Implemented comprehensive tests for the component's functionality and user interactions.

montelaidev committed:
fix: storybook links

owencraston commented on app/components/Views/MultichainAccounts/MultichainAccountConnect/MultichainAccountConnectMultiSelector/MultichainAccountConnectMultiSelector.tsx:
```diff
@@ -0,0 +1,195 @@
+  );
+
+  return (
+    <SafeAreaView>
```
The SafeAreaView from `react-native` is being deprecated and will cause us issues when we eventually update to the latest version. Instead we should use the SafeAreaView from `react-native-safe-area-context`

owencraston commented on app/components/Views/MultichainAccounts/MultichainAccountConnect/MultichainAccountConnectMultiSelector/MultichainAccountConnectMultiSelector.tsx:
```diff
@@ -0,0 +1,195 @@
+          <View style={styles.sdkInfoContainer}>
+            <View style={styles.sdkInfoDivier} />
+            <Text color={TextColor.Muted}>
+              SDK {connection?.originatorInfo?.platform} v
```
Should these be translated?

owencraston commented on app/components/Views/MultichainAccounts/MultichainAccountConnect/MultichainAccountConnectMultiSelector/MultichainAccountConnectMultiSelector.tsx:
```diff
@@ -0,0 +1,195 @@
+
+  const accountGroups = useSelector(selectAccountGroups);
+
+  const selectedAccountGroups = useMemo(
```
why are we building this list? The child component creates a set of selected accounts so could we not just pass in the state of `selectedAccountGroupIds` directly to the `MultichainAccountSelectorList`

montelaidev replied:
The selector list includes the wallet headers which aren't used here.

montelaidev committed:
refactor: update MultichainAccountConnectMultiSelector to use Set for selected account groups

- Replaced array state management with Set for selected account group IDs to improve performance and simplify logic.
- Updated mock internal account creation to utilize KeyringTypes for better clarity and consistency.
- Enhanced localization by adding a new string for SDK connection details in the language files.

montelaidev committed:
Merge branch 'main' into feat/multichain-account-conecct-multiselector

montelaidev commented:
Adding `skip-sonar-check` because the cta callback is a duplicate from the non bip44 AccountConnectMultiselector.
